include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -201,26 +196,52 @@ void assembleZVec(SizeType i_begin, SizeType i_split, SizeType i_end, pika::shar
     GlobalTileIndex mat_ev_idx(mat_ev_row, i);
     // Take the last row of a `Q1` tile or the first row of a `Q2` tile
     GlobalTileIndex z_idx(i, 0);
+
     // Copy the row into the column vector `z`
-    whenAllLift(top_tile, rho_fut, mat_ev.read_sender(mat_ev_idx), z.readwrite_sender(z_idx)) |
-        copyTileRowAndNormalize(Policy<Backend::MC>(thread_priority::normal)) | start_detached();
+    auto copy_fn = [top_tile](const auto& rho, const auto& tile, const auto& col,
+                              [[maybe_unused]] auto&&... ts) {
+      // Copy the bottom row of the top tile or the top row of the bottom tile
+      SizeType row = (top_tile) ? col.size().rows() - 1 : 0;
+
+      // Negate Q1's last row if rho < 0
+      //
+      // lapack 3.10.0, dlaed2.f, line 280 and 281
+      int sign = (top_tile && rho < 0) ? -1 : 1;
+
+      if constexpr (D == Device::CPU) {
+        for (SizeType i = 0; i < tile.size().cols(); ++i) {
+          col(TileElementIndex(i, 0)) = sign * tile(TileElementIndex(row, i)) / T(std::sqrt(2));
+        }
+      }
+      else {
+        copyTileRowAndNormalizeOnDevice(sign, tile.size().cols(), tile.ld(),
+                                        tile.ptr(TileElementIndex(row, 0)), col.ptr(), ts...);
+      }
+    };
```

Comment:
Same comment here about dispatching. Even adding a `copyTileRowAndNormalizeOnDevice` overload without a stream argument would be sufficient, i.e. to allow something like:
```suggestion
    auto copy_fn = [top_tile](const auto& rho, const auto& tile, const auto& col,
                              [[maybe_unused]] auto&&... ts) {
      // Copy the bottom row of the top tile or the top row of the bottom tile
      SizeType row = (top_tile) ? col.size().rows() - 1 : 0;

      // Negate Q1's last row if rho < 0
      //
      // lapack 3.10.0, dlaed2.f, line 280 and 281
      int sign = (top_tile && rho < 0) ? -1 : 1;
      copyTileRowAndNormalizeOnDevice(sign, tile.size().cols(), tile.ld(),
                                      tile.ptr(TileElementIndex(row, 0)), col.ptr(), ts...);
      }
    };
```
I think just adding non-stream overloads to the other places with the same comment would also be an option.

Reply:
Relevant issue: https://github.com/eth-cscs/DLA-Future/issues/634